lib/log.c

```diff
@@ -167,13 +172,21 @@ int log_set_logfile(const char *logfile) {
 			}
 			fp_logfile = fp;
 		}
+	}
+	return 0;
+}
+
+int log_set_logfile(const char *logfile_path) {
+	if (logfile_path == NULL) {
+		logfile = NULL;
 	} else {
-		if (fp_logfile) {
-			fclose(fp_logfile);
-			fp_logfile = NULL;
+		logfile = strdup(logfile_path);
```

Comment:
This will leak memory if the log file name is changed during program execution.
It's not a problem now, but I'd prefer that we guard against that.

Reply:
Added a `free`, please have another look.

Reply:
That part is good, thanks.